src/d3d8/d3d8_main.cpp

```diff
@@ -19,14 +19,38 @@ extern "C" {
       const D3DCAPS8*  pCaps,
       BOOL             errorReturn,
       char**           pErrorString) {
-    dxvk::Logger::warn("D3D8: ValidatePixelShader: Stub");
+    const char* errorMessage = "";
+
+    if (unlikely(pPixelShader == nullptr)) {
+      errorMessage = "D3D8: ValidatePixelShader: Null pPixelShader";
+    } else {
+      uint32_t majorVersion = (pPixelShader[0] >> 8) & 0xff;
+      uint32_t minorVersion = pPixelShader[0] & 0xff;
+
+      if (unlikely(majorVersion != 1 || minorVersion > 4)) {
+        errorMessage = dxvk::str::format("D3D8: ValidatePixelShader: Unsupported PS version ",

```

Comment:
This looks like UAF. The returned string is freed immediately but you're still its contents.

Reply:
Nice catch, thanks. Fixed.